modules/distribution/utils/delegator_rewards.go

```diff
@@ -13,8 +13,27 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// BlockInterval returns interval time in blocks
+var BlockInterval int64 = 100
+
 // UpdateDelegatorsRewardsAmounts updates the delegators commission amounts
 func UpdateDelegatorsRewardsAmounts(height int64, client distrtypes.QueryClient, db *database.Db) {
+	rewards, _ := db.GetDelegatorRewards()
+
+	if len(rewards) == 0 {
+		// run once when chain starts
+		go updateDelegatorsRewards(height, client, db)
+	}
+
+	if len(rewards) > 0 {
+		// run every 100 blocks interval
+		if height%BlockInterval == 0 {
+			go updateDelegatorsRewards(height, client, db)
+		}
+	}
```

Comment:
I think this can be simplified as follows: 

```suggestion
	if len(rewards) == 0 || height%BlockInterval == 0 {
		// run once when chain starts
		go updateDelegatorsRewards(height, client, db)
	}
```

Also, I think we should move the `BlockInterval` value somewhere else. Maybe we can set it inside the module's constructor method so that in other chains (e.g. `Osmosis`) we can get the value from the database when first creating the module?

Reply:
Maybe even a better alternative is to have a custom BDJuno params table inside the database that holds this values. This way we can even play with them to find the best value without pushing out new versions of BDJuno in the future

Reply:
BlockInterval value will always be hardcoded, so how about adding the value to bdjuno config file and saving it to distribution_params table from there?

Reply:
@MonikaCat Why is it hardcoded? I mean, we might want to update the delegations on Cosmos every 30 blocks and on Terra every 100. Also, on epoch-based chains like Osmosis the epoch interval will surely be present inside on-chain params

Reply:
I tried to query epoch length value for osmosis chain but I can't seem to find it

Reply:
Can you point me to the correct function for that?

Reply:
@MonikaCat I think you can get it by qierying the [`mint` params](https://github.com/osmosis-labs/osmosis/blob/cf3246aaad8c2269225b430a1c3dd776a172ba63/x/mint/spec/04_params.md). This contains a `epoch_identifier` that can be used to query the details of such epoch from the [`epoch` module](https://github.com/osmosis-labs/osmosis/blob/cf3246aaad8c2269225b430a1c3dd776a172ba63/x/epochs/spec/02_state.md).

Reply:
Cool! I updated osmosis PR #198 with querying ```epoch_identifier``` value instead of hard coding ```BlockInterval``` value.

Reply:
The base branch however will need to have this value hard coded because, as you mentioned already, we might want to adjust it for each chain individually.

Reply:
@MonikaCat I think it would be way better to extend the current configuration allowing us to set such value from there instead that from the code. Something like this would be perfect ideally: 

```toml
[distribution]
rewards_frequency = 100
```

This way, in the future we an also add different frequencies for other queries and modules as well. What do you think?

Reply:
Yup I agree! It would also help users to setup bdjuno config in one place, without the need to modify the source code. Alright I will update the code with the above :)

Reply:
I added the rewards_frequency field in config.toml file and updated the source code, what do you think?